core/ledger/ledger.go

```diff
@@ -159,7 +159,25 @@ func (ledger *Ledger) CommitTxBatch(id interface{}, transactions []*protos.Trans
 	writeBatch := gorocksdb.NewWriteBatch()
 	defer writeBatch.Destroy()
 	block := protos.NewBlock(transactions, metadata)
-	block.NonHashData = &protos.NonHashData{}
+
+	ccEvents := []*protos.ChaincodeEvent{}
+
+	if transactionResults != nil {
+		ccEvents = make([]*protos.ChaincodeEvent, len(transactionResults))
+		for i := 0; i < len(transactionResults); i++ {
+			if transactionResults[i].ChaincodeEvent != nil {
+				ccEvents[i] = transactionResults[i].ChaincodeEvent
+			} else {
+				//protobuf does not like elements of a repeated array to be nil
+				//we should discard empty events without chaincodeID when sending
+				//out events
+				ccEvents[i] = &protos.ChaincodeEvent{}
```

Comment:
Yes, you are right, you will lose indexes that way. Btw it is a bit 'weird' that we have to use empty ChaincodeEvents as null values (they are semantically different). I will try to find some better/nicer solution for the future but now this is OK.